e2e-tests/tests/committee/test_committee.py

```diff
@@ -3,37 +3,80 @@
 from src.pc_epoch_calculator import PartnerChainEpochCalculator
 from src.partner_chain_rpc import DParam
 from config.api_config import ApiConfig
-from pytest import mark, skip
+from pytest import fixture, mark, skip
 from src.db.models import Candidates, PermissionedCandidates, StakeDistributionCommittee
-from sqlalchemy import func, select
+from sqlalchemy import select
 from sqlalchemy.orm import Session
 import math
 import numpy as np
 
 
-def calculate_d_param_tolerance(pc_epochs_in_mc_epoch, d_param_p, d_param_t):
-    """
-    Calculate the tolerance for the committee ratio test.
-    The tolerance is calculated by running a large number of simulations
-    """
-    # Probability of p_cnt increment
-    p_prob = d_param_p / (d_param_p + d_param_t)
+class TestCommitteeDistribution:
+    @fixture(scope="class")
+    def p_candidates_available(self, api: BlockchainApi):
+        candidates = {}
 
-    # Simulate all dice rolls at once using NumPy's capabilities
-    dice_rolls = np.random.randint(1, d_param_p + d_param_t + 1, (50000, pc_epochs_in_mc_epoch))
-    p_counts = np.sum(dice_rolls <= d_param_p, axis=1)
+        def _p_candidates_available(mc_epoch):
+            if mc_epoch not in candidates:
+                candidates[mc_epoch] = api.get_permissioned_candidates(mc_epoch, valid_only=True)
+            return candidates[mc_epoch]
 
-    highest_cnt = np.max(p_counts)
-    observed_minimum_required_tolerance = highest_cnt / (p_prob * pc_epochs_in_mc_epoch) - 1
-    d_param_tolerance = 1.1 * observed_minimum_required_tolerance
-    logging.debug(
-        f"Observed minimum tolerance: {observed_minimum_required_tolerance}. Tolerance to use: {d_param_tolerance}."
-    )
+        return _p_candidates_available
 
-    return d_param_tolerance
+    @fixture(scope="class")
+    def p_candidates_count(
+        self,
+        current_mc_epoch,
+        get_pc_epoch_committee,
+        p_candidates_available,
+        pc_epoch_calculator: PartnerChainEpochCalculator,
+    ):
+        candidates_count = {}
 
+        def _p_candidates_count(pc_epoch):
```

Comment:
I don't think it returns candidates count in a pc_epoch. It returns number of committee seats assigned to permissioned candidates. Committee size is not same as PC epoch length.

Reply:
I think it is a matter of rename or doc comment. Reading further I understood it is about committee of pc_epoch.